arrows/qt/image_io.cxx

```diff
@@ -0,0 +1,132 @@
+/*ckwg +29
+ * Copyright 2018 by Kitware, Inc.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ *  * Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ *
+ *  * Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ *  * Neither name of Kitware, Inc. nor the names of any contributors may be used
+ *    to endorse or promote products derived from this software without specific
+ *    prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE AUTHORS OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * \file
+ * \brief Qt image_io implementation
+ */
+
+#include "image_io.h"
+
+#include <arrows/qt/image_container.h>
+
+namespace {
+
+// ----------------------------------------------------------------------------
+QString
+qt_string( std::string const& in )
+{
+  return QString::fromLocal8Bit( in.data(), static_cast< int >( in.size() ) );
+}
+
+} // end namespace (anonymous)
+
+namespace kwiver {
+
+namespace arrows {
+
+namespace qt {
+
+// ----------------------------------------------------------------------------
+image_io
+::image_io()
+{
+  attach_logger( "arrows.qt.image_io" );
+}
+
+// ----------------------------------------------------------------------------
+image_io
+::~image_io()
+{
+}
+
+// ----------------------------------------------------------------------------
+vital::config_block_sptr
+image_io
+::get_configuration() const
+{
+  return {};
```

Comment:
The return here is returning a NULL sptr.  The proper return, I believe, is config_block::empty_config() which is returned by the parent class.  Probably best not to override here.

Regardless, returning a NULL ptr here is causes a segfault in plugin_explorer.

Reply:
Ugh, `set_configuration` and `check_configuration` are pure virtual, but *not* `get_configuration`? Lovely.